src/main/java/io/ballerina/plugins/idea/configuration/ui/BallerinaSdkSelection.java

```diff
@@ -0,0 +1,310 @@
+                selectedVersionTextField.setText(savedVersion);
+                selectedSdkPath = savedPath;
+                selectedSdkVersion = savedVersion;
+            } else {
+                if (BallerinaSdkUtil.isValidSdk(systemBalPath, systemBalVersion)) {
+                    sdkVersionComboBox.setSelectedIndex(0);
+                    selectedVersionTextField.setText(systemBallerinaVersionStart + systemBalVersion);
+                    selectedSdkPath = systemBalPath;
+                    selectedSdkVersion = systemBalVersion;
+                } else {
+                    selectedVersionTextField.setText(noValidBalSdk);
+                    selectedVersionTextField.setForeground(JBColor.RED);
+                    selectedSdkVersion = "";
+                    selectedSdkPath = "";
+                }
+            }
+        }
+    }
+
+    private void fillComboBox() {
+        for (BallerinaSdkUtil.BallerinaSdk sdk : sdkList) {
+            if (!Objects.equals(selectedSdkPath, sdk.getPath()) && !Objects.equals(systemBalPath, sdk.getPath())) {
+                sdkVersionComboBox.addItem(BallerinaSdkUtil.findBalDistFolder(sdk.getPath()));
+            }
+        }
+        sdkVersionComboBox.addItem(addBallerinaSdk);
+        sdkVersionComboBox.setEnabled(useCustomSdkCheckbox.isSelected());
+    }
+
+    private void setupCheckBoxListener() {
+        useCustomSdkCheckbox.addActionListener(e -> {
+            boolean selected = useCustomSdkCheckbox.isSelected();
+            if (!selected) {
+                String path = sdkVersionComboBox.getItemAt(0);
+                if (Objects.equals(path, comboBoxDefaultStart + BallerinaSdkUtil.findBalDistFolder(systemBalPath))) {
+                    sdkVersionComboBox.setSelectedIndex(0);
+                    selectedVersionTextField.setText(systemBallerinaVersionStart + systemBalVersion);
+                    selectedSdkVersion = systemBalVersion;
+                    selectedSdkPath = systemBalPath;
+                } else {
+                    sdkVersionComboBox.setSelectedItem(addBallerinaSdk);
+                    selectedVersionTextField.setText(noValidBalSdk);
+                    selectedSdkVersion = "";
+                    selectedSdkPath = "";
+                }
+            }
+            sdkVersionComboBox.setEnabled(selected);
+        });
+    }
+
+    private void setupComboBoxListener() {
+        sdkVersionComboBox.addActionListener(e -> {
+            if (addBallerinaSdk.equals(sdkVersionComboBox.getSelectedItem())) {
+                handleCustomFolderSelection();
+            } else {
+                handleDetectedPathsSelection();
+            }
+        });
+    }
+
+    private void handleCustomFolderSelection() {
+        VirtualFile file =
+                FileChooser.chooseFile(new FileChooserDescriptor(false, true,
+                                false, false,
+                                false, false),
+                        project, null);
```

Comment:
```suggestion
        VirtualFile file = FileChooser.chooseFile(
                               new FileChooserDescriptor(false, true, false, false, false, false), project, null);
```

Reply:
This change would cause the character count to exceed the 120-character limit.